autoload/lsp/diag.vim

```diff
@@ -211,6 +211,11 @@ enddef
 # Notification: textDocument/publishDiagnostics
 # Param: PublishDiagnosticsParams
 export def DiagNotification(lspserver: dict<any>, uri: string, newDiags: list<dict<any>>): void
+  # Diagnostics are disabled for this server
+  if lspserver.features->has_key('diagnostics') && !lspserver.features.diagnostics
+    return
+  endif
+
```

Comment:
I don't think there is any way in the specifications to tell a language server NOT to send diagnostics, so they are ignored when they are received instead.